app-android/src/main/java/com/ustadmobile/port/android/view/ClazzEditFragment.kt

```diff
@@ -284,4 +317,376 @@ class ClazzEditFragment() : UstadEditFragment<ClazzWithHolidayCalendarAndSchoolA
+            Text(text = stringResource(id = R.string.course_blocks))
+        }
+
+        item {
+            ListItem(
+                modifier = Modifier.clickable {
+                    onClickAddCourseBlock()
+                },
+                text = { Text(stringResource(id = R.string.add_block)) },
+            )
+        }
+
+        items (
+            items = uiState.courseBlockList
+        ) { courseBlock ->
+
+            val CONTENT_ENTRY_TYPE_ICON_MAP = mapOf(
+                ContentEntry.TYPE_EBOOK to R.drawable.ic_book_black_24dp,
+                ContentEntry.TYPE_VIDEO to R.drawable.video_youtube,
+                ContentEntry.TYPE_DOCUMENT to R.drawable.text_doc_24px,
+                ContentEntry.TYPE_ARTICLE to R.drawable.article_24px,
+                ContentEntry.TYPE_COLLECTION to R.drawable.collections_24px,
+                ContentEntry.TYPE_INTERACTIVE_EXERCISE to R.drawable.ic_baseline_touch_app_24,
+                ContentEntry.TYPE_AUDIO to R.drawable.ic_audiotrack_24px
+            )
+            val image = if(courseBlock.cbType == CourseBlock.BLOCK_CONTENT_TYPE)
+                courseBlock.entry?.contentTypeFlag
+            else
+                courseBlock.cbType
+
+            val courseBlockEditAlpha: Float = if (courseBlock.cbHidden) 0.5f else 1f
+
+            ListItem(
+                modifier = Modifier.alpha(courseBlockEditAlpha),
+                icon = {
+                    Row{
+                        Icon(
+                            Icons.Filled.Menu,
+                            contentDescription = null
+                        )
+                        Icon(
+                            painterResource(id = CONTENT_ENTRY_TYPE_ICON_MAP[image] ?: R.drawable.text_doc_24px),
+                            contentDescription = null
+                        )
+                    }
+                },
+
+                text = { Text(courseBlock.cbTitle ?: "") },
+                trailing = {
+                    PopUpMenu(
+                        courseBlock,
+                        onClickHideBlockPopupMenu,
+                        onClickIndentBlockPopupMenu,
+                        onClickUnIndentBlockPopupMenu,
+                        onClickDeleteBlockPopupMenu,
+                    )
+                }
+            )
+        }
+
+        item {
+            Spacer(modifier = Modifier.height(20.dp))
+        }
+
+        item {
+            Text(text = stringResource(id = R.string.schedule))
+        }
+
+        item {
+            ListItem(
+                modifier = Modifier.clickable {
+                    onClickAddSchedule()
+                },
+                text = { Text(stringResource(id = R.string.add_a_schedule)) },
+                icon = {
+                    Icon(
+                        imageVector = Icons.Filled.Add,
+                        contentDescription = "",
+                    )
+                }
+            )
+        }
+
+        items(
```

Comment:
better to use:
```
items(
 items = uiState.clazzSchedules
) { schedule ->
   
}
```

We don't need to access items by using an array index. I think this is clearer than using it as an index.

Also: the Schedules are part of a List, so they should use ListItem, not UstadDetailField.